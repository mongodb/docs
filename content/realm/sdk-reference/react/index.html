<!doctype html>
<head>
  <meta http-equiv="refresh" content="0; url=https://mongodb.com/docs/realm-sdks/react/latest" />
</head>
<body>
  <p><a href="https://mongodb.com/docs/realm-sdks/react/latest">Redirect</a></p>
</body>
