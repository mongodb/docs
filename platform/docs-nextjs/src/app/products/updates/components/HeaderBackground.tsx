import { theme } from '@/styles/theme';
import { css, cx } from '@leafygreen-ui/emotion';
import { palette } from '@leafygreen-ui/palette';
import Image from 'next/image';

const headerBackgroundStyle = css`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: ${palette.green.light3};
  border-radius: 0 0 40px 40px;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;

  /* Tablet breakpoint: between small and medium */
  @media ${theme.screenSize.mediumAndUp} {
    height: 230px;
    border-radius: 0 0 80px 80px;
  }

  /* Desktop breakpoint: larger than medium */
  @media ${theme.screenSize.largeAndUp} {
    height: 384px;
    border-radius: 0 0 80px 80px;
  }
`;

const leftImageStyle = css`
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;

  @media ${theme.screenSize.mediumAndUp} {
    top: 10px;
  }

  @media ${theme.screenSize['2XLargeAndUp']} {
    top: 0;
  }
`;

const rightImageStyle = css`
  position: absolute;
  top: 0px;
  right: 0px;
  z-index: 1;

  @media ${theme.screenSize.mediumAndUp} {
    top: 0px;
    right: -10px;
  }

  @media ${theme.screenSize['2XLargeAndUp']} {
    top: -20px;
    right: -20px;
  }
`;

const mobileImageStyle = css`
  display: block;

  @media ${theme.screenSize.mediumAndUp} {
    display: none;
  }
`;

const tabletImageStyle = css`
  display: none;

  @media ${theme.screenSize.mediumAndUp} {
    display: block;
  }

  @media ${theme.screenSize['2XLargeAndUp']} {
    display: none;
  }
`;

const desktopImageStyle = css`
  display: none;

  @media ${theme.screenSize['2XLargeAndUp']} {
    display: block;
  }
`;

const HeaderBackground = () => {
  return (
    <div className={cx(headerBackgroundStyle)}>
      {/* Left image container */}
      <div className={cx(leftImageStyle)}>
        <Image
          src="/product-updates/header-mobile-left.svg"
          alt="header-left"
          width={220}
          height={220}
          className={cx(mobileImageStyle)}
        />
        <Image
          src="/product-updates/header-tablet-left.svg"
          alt="header-left"
          width={320}
          height={320}
          className={cx(tabletImageStyle)}
        />
        <Image
          src="/product-updates/header-desktop-left.svg"
          alt="header-left"
          width={300}
          height={300}
          className={cx(desktopImageStyle)}
        />
      </div>

      {/* Right image container */}
      <div className={cx(rightImageStyle)}>
        <Image
          src="/product-updates/header-mobile-right.svg"
          alt="header-right"
          width={160}
          height={160}
          className={cx(mobileImageStyle)}
        />
        <Image
          src="/product-updates/header-tablet-right.svg"
          alt="header-right"
          width={200}
          height={200}
          className={cx(tabletImageStyle)}
        />
        <Image
          src="/product-updates/header-desktop-right.svg"
          alt="header-right"
          width={360}
          height={360}
          className={cx(desktopImageStyle)}
        />
      </div>
    </div>
  );
};

export default HeaderBackground;
