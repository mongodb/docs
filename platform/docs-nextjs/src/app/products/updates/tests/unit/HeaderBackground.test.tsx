import { render } from '@testing-library/react';
import HeaderBackground from '../../components/HeaderBackground';

describe('HeaderBackground component', () => {
  it('renders correctly', () => {
    const wrapper = render(<HeaderBackground />);
    expect(wrapper.queryByTestId('header-background-img')).toBeInTheDocument();
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
