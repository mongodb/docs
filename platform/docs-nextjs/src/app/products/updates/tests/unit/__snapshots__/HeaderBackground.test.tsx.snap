// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HeaderBackground component renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: #E3FCF7;
  border-radius: 0 0 40px 40px;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

@media not all and (max-width: 767px) {
  .emotion-0 {
    height: 230px;
    border-radius: 0 0 80px 80px;
  }
}

@media not all and (max-width: 1024px) {
  .emotion-0 {
    height: 384px;
    border-radius: 0 0 80px 80px;
  }
}

.emotion-1 {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
}

@media not all and (max-width: 767px) {
  .emotion-1 {
    top: 10px;
  }
}

@media not all and (max-width: 1440px) {
  .emotion-1 {
    top: 0;
  }
}

.emotion-2 {
  display: block;
}

@media not all and (max-width: 767px) {
  .emotion-2 {
    display: none;
  }
}

.emotion-3 {
  display: none;
}

@media not all and (max-width: 767px) {
  .emotion-3 {
    display: block;
  }
}

@media not all and (max-width: 1440px) {
  .emotion-3 {
    display: none;
  }
}

.emotion-4 {
  display: none;
}

@media not all and (max-width: 1440px) {
  .emotion-4 {
    display: block;
  }
}

.emotion-5 {
  position: absolute;
  top: 0px;
  right: 0px;
  z-index: 1;
}

@media not all and (max-width: 767px) {
  .emotion-5 {
    top: 0px;
    right: -10px;
  }
}

@media not all and (max-width: 1440px) {
  .emotion-5 {
    top: -20px;
    right: -20px;
  }
}

<div
    class="emotion-0"
    data-testid="header-background-img"
  >
    <div
      class="emotion-1"
    >
      <img
        alt="header-left"
        class="emotion-2"
        data-nimg="1"
        decoding="async"
        height="220"
        loading="lazy"
        src="/products/updates/header-mobile-left.svg"
        style="color: transparent;"
        width="220"
      />
      <img
        alt="header-left"
        class="emotion-3"
        data-nimg="1"
        decoding="async"
        height="320"
        loading="lazy"
        src="/products/updates/header-tablet-left.svg"
        style="color: transparent;"
        width="320"
      />
      <img
        alt="header-left"
        class="emotion-4"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/products/updates/header-desktop-left.svg"
        style="color: transparent;"
        width="300"
      />
    </div>
    <div
      class="emotion-5"
    >
      <img
        alt="header-right"
        class="emotion-2"
        data-nimg="1"
        decoding="async"
        height="160"
        loading="lazy"
        src="/products/updates/header-mobile-right.svg"
        style="color: transparent;"
        width="160"
      />
      <img
        alt="header-right"
        class="emotion-3"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/products/updates/header-tablet-right.svg"
        style="color: transparent;"
        width="200"
      />
      <img
        alt="header-right"
        class="emotion-4"
        data-nimg="1"
        decoding="async"
        height="360"
        loading="lazy"
        src="/products/updates/header-desktop-right.svg"
        style="color: transparent;"
        width="360"
      />
    </div>
  </div>
</DocumentFragment>
`;
